src/tests/ftest/util/thread_manager.py

```diff
@@ -0,0 +1,163 @@
+        """Return the string respresentation of this object.
+
+        Returns:
+            str: the string respresentation of this object
+
+        """
+        def get_result(result):
+            """Get the result information to display.
+
+            Args:
+                result (object): the result returned by the thread
+
+            Returns:
+                list: each line of the result information
+
+            """
+            data = []
+            if isinstance(result, CmdResult):
+                data.append("    command:     {}".format(result.command))
+                data.append("    exit_status: {}".format(result.exit_status))
+                data.append("    duration:    {}".format(result.duration))
+                data.append("    interrupted: {}".format(result.interrupted))
+                data.append("    stdout:")
+                for line in result.stdout_text.splitlines():
+                    data.append("      {}".format(line))
+                data.append("    stderr:")
+                for line in result.stderr_text.splitlines():
+                    data.append("      {}".format(line))
+            else:
+                for line in str(result).splitlines():
+                    data.append("    {}".format(line))
+            return data
+
+        info = ["Thread {} results:".format(self.id), "  args: {}".format(self.args), "  result:"]
+        if isinstance(self.result, list):
+            for this_result in self.result:
+                info.extend(get_result(this_result))
+        else:
+            info.extend(get_result(self.result))
+        return "\n".join(info)
+
+
+class ThreadManager():
+    """Class to manage running any method as multiple threads."""
+
+    def __init__(self, method, timeout=None):
+        """Initialize a ThreadManager object with the the method to run as a thread.
+
+        Args:
+            method (callable): [description]
+            timeout (int, optional): [description]. Defaults to None.
+        """
+        self.log = getLogger()
+        self.method = method
+        self.timeout = timeout
+        self.job_kwargs = []
+        self.futures = {}
+
+    @property
+    def qty(self):
+        """Get the number of threads.
+
+        Returns:
+            int: number of threads
+
+        """
+        return len(self.job_kwargs)
+
+    def add(self, **kwargs):
+        """Add a thread to run by specifying the keyword arguments for the thread method."""
+        self.job_kwargs.append(kwargs)
+
+    def run(self):
+        """Asynchronously run the method as a thread for each set of method arguments.
+
+        Returns:
+            list: a list of ThreadResults for the execution of each method.
+
+        """
+        results = []
+        with ThreadPoolExecutor() as thread_executor:
+            self.log.info("Submitting %d threads ...", len(self.job_kwargs))
+            futures = {
+                thread_executor.submit(self.method, **kwargs): index
```

Comment:
How is this dictionary created? Create a comment. Do we need a comprehension type of assignment for creating this dictionary? I would make is simple so it is easy to understand...
I believe we are creating a futures dictionary like:
{thread_executer.submit(x, Xargs): index, thread_executer.submit(Y, Yargs): index , so on...}